python/ray/services.py

```diff
@@ -101,12 +102,25 @@ def find_redis_address_or_die():
     redis_addresses = set()
     for pid in pids:
         try:
-            proc = psutil.Process(pid)
-            for arglist in proc.cmdline():
-                for arg in arglist.split(" "):
-                    if arg.startswith("--redis-address="):
-                        addr = arg.split("=")[1]
-                        redis_addresses.add(addr)
+            cmdline = psutil.Process(pid).cmdline()
+
+            # (Workaround for UNIX idiosyncrasy)
+            # See: https://github.com/giampaolo/psutil/issues/1179
+            # Normally, cmdline() should work. But it seems that in some cases
+            # (due to setproctitle) the entire command-line is stored into
+            # the first argument, and lots of empty arguments follow.
+            # Explained here: https://unix.stackexchange.com/a/432681
+            # We try to detect this case and split manually ONLY in this case.
+            suspect = len(cmdline) > 1 and all(not arg for arg in cmdline[1:])
+            if suspect and not sys.platform.startswith("win32"):
+                # This can be incorrect (due to lack of quoting), but we can't
+                # do better. And in this _particular_ use case, this is
+                # unlikely to be problematic.
+                cmdline = shlex.split(cmdline[0])
```

Comment:
We're just looking for redis_address=N right? Why would this ever have an effect?

Reply:
Normally the correct thing to do is to go through each command-line argument and check to see if its prefix is what we expect. I'm fixing that here. But if `setproctitle` has been called, then we get this weird behavior that I'm providing a fallback for and documenting.